config_parser.cc

```diff
@@ -200,13 +202,21 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
       config_stack.push(new_config);
     } else if (token_type == TOKEN_TYPE_END_BLOCK) {
       if (last_token_type != TOKEN_TYPE_STATEMENT_END) {
-        // Error.
-        break;
+        if (last_token_type == TOKEN_TYPE_END_BLOCK) {
+          bracketCounter--;
+        }
+        else {
+         // Error.
+         break;
+        }
+      }
+      else {
+        bracketCounter--;
```

Comment:
I think having an extra if else makes it sort of confusing. If you just have your brackerCounter decrement right away once you enter TOKEN_TYPE_END_BLOCK, and change the original if statement to also check that the last_token_type != TOKEN_TYPE_END_BLOCK before breaking, then it'll be a little more straight forward. 

Right now I feel it's a little confusing why you have bracketCounter-- in two different places. It doesn't flow that you would decrement if the last token is an end block, although I understand now that you're trying to fix the bug with multiple nested brackets.